referenceframe/frame_test.go

```diff
@@ -77,10 +77,11 @@ func TestPrismaticFrame(t *testing.T) {
 
 	randomInputs := RandomFrameInputs(frame, nil)
 	test.That(t, len(randomInputs), test.ShouldEqual, len(frame.DoF()))
-	restrictRandomInputs := RestrictedRandomFrameInputs(frame, nil, 0.001)
+	restrictRandomInputs, err := RestrictedRandomFrameInputs(frame, nil, 0.001, FloatsToInputs([]float64{-10}))
+	test.That(t, err, test.ShouldBeNil)
 	test.That(t, len(restrictRandomInputs), test.ShouldEqual, len(frame.DoF()))
-	test.That(t, restrictRandomInputs[0].Value, test.ShouldBeLessThan, 0.03)
-	test.That(t, restrictRandomInputs[0].Value, test.ShouldBeGreaterThan, -0.03)
+	test.That(t, restrictRandomInputs[0].Value, test.ShouldBeLessThan, -9.07)
+	test.That(t, restrictRandomInputs[0].Value, test.ShouldBeGreaterThan, -10.03)
```

Comment:
This was short sighted in my original test but we should probably produce and test multiple numbers here to show that it wasn't just lucky.  So can just stick this whole chunk in a for loop that counts to 10 as in the function below